src/common/Thread.h

```diff
@@ -56,4 +58,17 @@ class Thread {
   int set_affinity(int cpuid);
 };
 
+template<typename Fun, typename... Args>
+std::thread make_named_thread(const std::string& s,
+			      Fun&& fun,
+			      Args&& ...args) {
+  auto t = std::thread(std::forward<Fun>(fun),
+		       std::forward<Args>(args)...);
+  int r = pthread_setname_np(t.native_handle(), s.c_str());
```

Comment:
@adamemerson this breaks the build at least on FreeBSD and Solaris. FreeBSD has `pthread_set_name_np()`. see `ceph_pthread_setname()` in `compat.h`.